addon-chrome/ChromePreviewProvider.js

```diff
@@ -0,0 +1,142 @@
+const db = require("addon-chrome/db");
+const {METADATA} = require("addon-chrome/constants");
+const {getMetadata} = require("page-metadata-parser");
+
+module.exports = class ChromePreviewProvider {
+   /**
+   * Fetches metadata for a list of sites
+   *
+   * @param {Array} sites - List of sites to get metadata for
+   * @returns {Object} Promise that resolves with list of sites with their metadata
+   */
+  static getLinksMetadata(sites) {
+    return new Promise((resolve) => {
+      const metadataPromise = sites
+        .filter((site) => !/(google|localhost|chrome)/.test(site.url))
+        .slice(0, 8)
```

Comment:
why specifically 8? make it into a constant, and comment how it was chosen, and the obvious reason of why we need to limit